### DISCUSSION TITLE
Controller neighbors missing nodes?

### DISCUSSION THREAD

**Discussion by nervling** [zwave-js/zwave-js-ui#2640] [Q&A]
Not filing this as a bug, as I don't understand if this is WAI or not (but I don't think it is...). The TLDR is my controller has 3 neighbors (far away) despite ~30+ nodes having the controller listed as a neighbor. Is there a way to force the controller to update which nodes are its neighbors? I've tried calling the MQTT refreshNeighbors API, healing the network, etc, but the controller doesn't seem to update its neighbor list for some reason. I'm running a Aeotec 7 stick, so this could be an issue with the buggy 700 firmware.

A very small snippet of the neighbor list is below. You can see 2, 7, and 8 all include 1, but 1 does not include any of them.

`{"1":[39,47,80],"2":[1,5,7,9,10,13,14,15,16,17,18,19,20,21,22,23,24,26,27,30,36,40,41,47,48,50,51,52,61,62,64,65,66,69,71,73,75,79,80,83,87],"5":[2,7,8,9,10,11,12,16,17,21,22,25,26,30,32,39,42,43,44,45,47,53,61,62,64,67,71,72,73,74,79],"7":[1,2,5,8,9,10,11,12,13,15,16,17,18,19,20,21,22,24,26,30,32,39,40,42,43,44,45,48,53,61,64,71,72,73,74,79],"8":[1,5,7,9,10,11,12,13,15,16,18,19,20,21,22,24,25,26,30,32,39,42,43,44,45,47,53,61,71,72,74,79,89]`

**Comment by robertsLando** [comment]
@nervling did you tried to run healthcheck on it? Also check this: https://zwave-js.github.io/zwavejs2mqtt/#/usage/nodes_healthcheck

cc @AlCalzone

**Reply by AlCalzone** [reply]
Is the stick on an USB extension cord?
If yes, don't worry about it too much, as long as everything works.

**Reply by nervling** [reply]
No extension cord. I have a beelink t4 pro running docker + zwavejs2mqtt in my Kitchen.

I hadn't, included the results of one below. They all seem to be 7/10, somtimes the 2nd+ rounds are failing. The one included below is < 10 feet in clear line of sight of the controller and is still 7/10, which is the same I got for one of the single hop nodes that's downstairs and much further away.

`[
  {
    "latency": 30,
    "failedPingsNode": 0,
    "numNeighbors": 36,
    "routeChanges": 0,
    "snrMargin": 14,
    "rating": 7,
    "minPowerlevel": 1
  },
  {
    "latency": 50,
    "failedPingsNode": 0,
    "numNeighbors": 36,
    "routeChanges": 0,
    "snrMargin": 14,
    "rating": 7,
    "minPowerlevel": 4
  }
]`

**Reply by AlCalzone** [reply]
Use a USB extension. It's the very first troubleshooting step we recommend: https://zwave-js.github.io/node-zwave-js/#/troubleshooting/connectivity-issues?id=general-troubleshooting

**Reply by nervling** [reply]
Okay, you think it may help even though it's a very small computer not near anything else? To be clear I'd only be doing it to move it farther from the computer, its location is fine as is otherwise.

**Reply by AlCalzone** [reply]
Yes. The USB ports themselves are the problem. I literally can't include or exclude devices when my USB Stick is put directly into my laptop. Controlling them is a game of chance.

**Reply by nervling** [reply]
Got it, I'll give that a try and report back. Is your theory that may be what's causing the incorrect neighbors on the controller? I'm not wrong in that the others should be showing up as 1-hop neighbors right? If this doesn't work I think I'll try downgrading to a gen5+ aoetec stick, so many complaints about the 700 series.

**Reply by AlCalzone** [reply]
I'm not 100% sure what may be causing the incorrect neighbors, but I've had reports where using a USB extension fixes it (and some where it didn't).
It is guaranteed to improve connectivity a lot, or at least prevent issues.

**Reply by nervling** [reply]
Great. Thanks for being so responsive. The extension is coming from Amazon tomorrow, I'll update here once I know more.

**Reply by nervling** [reply]
Unfortunately same behavior. I've ordered a gen5+ stick that I'm going to try and copy my configuration to and see if it fixes anything.

**Reply by AlCalzone** [reply]
I wouldn't be too focused on the neighbor information - if everything works fine. It might also take some time and network heals to "fix" itself.

**Reply by nervling** [reply]
Yea, it's been like this for awhile and it doesn't work well after ~2 years of being fine. Node show up dead and need to be re-pinged often, including ones right next to the controller. It started when I moved my gen5 into a different server, i migrated to the gen7 thinking the stick was dying but it made things worse potentially if anything.

**Reply by AlCalzone** [reply]
Ok then it's time to use driver logs to investigate.
Start [here](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index), work through the logs and compare with [this](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/connectivity-issues) to identify what is failing how.
If this is actually caused by nodes having an issue, [this page](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/network-health) has some steps you should try.

If you're unsure with the logs, I can take a look.

**Reply by nervling** [reply]
I backed up my Gen7 aeotec stick config, restored to a Gen5+, and healed. So far everything (including the neighbor issue) seems to be fixed and working more reliably.

I've read ALOT of complaints online about the 700 series drivers being buggy, so I think that's what was going on here.

**Reply by AlCalzone** [reply]
Hmm, this might be another case of the Aeotec Gen7 having range issues.

**Comment by nervling** [comment]
Totally possible. I have 68 devices spread across a two floor 9000 square foot home. I did have devices failing that were within 20 feet of the controller though. Everything immediately went live after a ping. 0 dead devices since switching to gen5+.
